c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -198,6 +198,8 @@ struct THCCachingAllocator
       block = *it;
       free_blocks.erase(it);
     } else {
+      // TODO: We can check if some blocks become available by calling torch::CudaIPCCollect,
+      // but right now such linking will require lots of code changes.
```

Comment:
Not that many code changes. All you need is to add a user settable callback function "call me when you run out of memory", and then libtorch can register the "run IPC collection" hook here. So it's a little more code, but not "lots of code changes"

Reply:
Callbacks implemented